fix realtek_rtl8195am IPv6 build fails

M-ichae-l opened the pull request:
### Description

- remove the "dhcps.h" and "dhcps.c" files
- edit "wifi_conf.c" and "rtw_emac.cpp" to fix compile error
- coding style fix


### Pull request type

    [X] Fix
    [ ] Refactor
    [ ] New target
    [ ] Feature
    [ ] Breaking change

c1728p9 committed:
Add documentation and test the HAL RTC API

Add requirements, tests, an example implementation and additional
function documentation to the HAL RTC API.

0xc0170 committed:
Sleep: add time requirements for sleep

Sleep - within 10us
Deepsleep - within 10ms

Note about mbed boards with interface, moved to lpc176x, as they are target related,
should be documented in the target documentation.

The tests will come as separate PR, to conform to this updates to sleep API.

c1728p9 committed:
Add documentation for the HAL Ticker API

Add documentation and test header for the HAL Ticker API.

mprse committed:
Add tests for ticker HAL API.

c1728p9 committed:
Turn off RTC until updated

Turn off RTC for all devices. When support for a device is added this
should be re-enabled.

c1728p9 committed:
Keep rtc code if low power timer is used

Keep the RTC code if either DEVICE_RTC or DEVICE_LOWPOWERTIMER is
defined on the devices which use the RTC for both the rtc api and the
low power timer api. This allows DEVICE_LOWPOWERTIMER to be enabled while
DEVICE_RTC is turned off.

c1728p9 committed:
Keep RTC API prototypes even when not enabled

Keep the prototypes in rtc_api.h even when DEVICE_RTC is not defined.
This allows devices that aren't fully compliant with the RTC API to
still use the header and prototypes.

mprse committed:
Add minor modifications of lp/us tickers tests desctiption in ticker header files.

Add one potential bug to the test description.
Fix test function names to be consistant with desctiption and test header.
Fix typos.

mprse committed:
Modify rtc_api.h header file - update tests specification.

mprse committed:
Update of RTC HAL API tests.

Provide minor fixes for existing test cases.
Add additional test cases.

mprse committed:
Add Sleep test specification to sleep_api.h.

mprse committed:
Add test and test header to Sleep HAL API.

bulislaw committed:
Add clarification to rtc_free doxygen

mprse committed:
NRF5 - make us ticker to be driven by high speed 1MHz timer

According to new ticker standards the following requirements for us ticker are not met on RRF5 boards:
- has a frequency between 250KHz and 8MHz (currently is driven by 32kHz clock)
- ticker increments by 1 each tick (currently is scaled to 1 MHz by incrementing counter by ~31)

Since BLE softdevice uses TIMER0 the proposition is to use high speed TIMER1 for us ticker configured as follows:
- TIMER counter width: 16 bits (max)
- TIMER frequency: 1MHz
This solution also uses Timer's capture/compare register 0 to specify interrupt time and Timer's capture/compare register 1 to read current timer value.

mprse committed:
Disable boards which does not fulfil new ticker standards.

mprse committed:
Disable boards which does not fulfil new sleep standards.

These boards will be re-enabled when sleep driver for them is ready.

Note:
This operation is done by removing "SLEEP" feature from target's "device_has" list (in targets.json config file).
For NRF52_DK removing of "SLEEP" feature causes some timing issues which have influence on tests. In order to successfully disable this board we need to disable also related features like "USTICKER", "LOWPOERTIMER" and slightly modify ticker tests, so they will not be executed if usticker support is not available (by default all targets support us ticker).

mprse committed:
Add DEVICE_RTC symbol check in RTC source files for NCS36510 target.

mprse committed:
Skip higher level ticker tests for targets with stale ticker target specific drivers.

Since target specific ticker drivers are not ready also features which uses ticker in upper layers may not work correctly and tests for these features.
We need to disable also failing higher level ticker related tests (by adding check if DEVICE_USTICKER symbol is available and raise #error [NOT_SUPPORTED] if not).

Commit:
Add lp/us ticker free() function implementation for NRF5

mprse committed:
Enable us/lp ticker support for NRF51_DK board

mprse committed:
Modify requirement for the ticker_init function.

Current version:
The function ticker_init resets the internal count and disables the ticker interrupt.

Proposed version:
The function ticker_init allows the ticker to keep counting and disables the ticker interrupt.

This is a result of the following discussion:
https://github.com/ARMmbed/mbed-os/pull/5233#pullrequestreview-86677815

mprse committed:
Modify ticker_init_test() to be consistent with the new requirement.

mprse committed:
Provide fix to the implementation of ticker_overflow_test test case.

The intention was to use ticker_overflow_delta equal to 0 for low power ticker tests and ticker_overflow_delta equal to 50 for high frequency ticker tests. Current implementation is invalid since for devices which provide LOW_POWER_TIMER feature delta is equal to 0 and for devices without this feature delta is equal 50.

mprse committed:
Update lp ticker to be consistent with the new standards.

Provide the following modifications for lp ticker driver:
- According to NRF51_DK reference manual rtc interrupt cannot be controlled by rtc event. In the previous implementation interrupts were enabled permanently and specific interrupt was enabled/disabled by enabling/disabling the specific event. If event is enabled, then event signal is provided to Programmable Peripheral Interconnect (PPI). If interrupt is enabled, then interrupt signal is provided to Nested Vector Interrupt Controller (NVIC). Disable all events permanently. Enable lp ticker overflow interrupt permanently(needed for RTC), disable lp ticker capture/compare interrupt on init (lp_ticker_init) , enable lp ticker interrupt when lp ticker interrupt is set (lp_ticker_set_interrupt), disable lp ticker interrupt on disable request(lp_ticker_disable_interrupt).
- Provide lp ticker data for higher level (freq: 32kHz / len: 24 bits),
- Add the following features to init routine: disable lp ticker interrupt.
- Make ticker driver to operate on ticks instead of us.
- Simplify lp ticker read and set interrupt routines (upper layers handle conversion to us and interrupt scheduling).

Commit:
Add free function to lp/us ticker HAL API

mprse committed:
Enable ticker support for K64F board.

mprse committed:
Adapt K64F us ticker driver to the new standards.

- provide ticker configuration: 1MHz/32 bit counter.
- us_ticker_init() routine disables interrupts.
- adapt comments.

mprse committed:
Adapt K64F lp ticker driver to the new standards.

Low power ticker time counter is created based on RTC which is driven by 32KHz clock. Additional low power timer is used to generate interrupts.
We need to adapt driver to operate on ticks instead of us.

Perform the following updates:
- provide lp ticker configuration: 32KHz/32 bit counter.
- lp_ticker_init() routine disables lp ticker interrupts .
- adapt the driver functions to operate on ticks instead us.
- adapt comments.
- add us_ticker_free() routine.

mprse committed:
lp_us_tickers test - provide counter overflow protection

Since according to the ticker requirements min acceptable counter size is 12 bits (low power timer) for which max count is 4095, then all test cases must be executed in this time window.
HAL ticker layer handles overflow and it is not handled in the target ticker drivers.

mprse committed:
lp_us_tickers test - add tolerance to interrupt time.

On some platforms (e.g. K64F) different counters are used for time measurement and interrupt generation.
Because of that we should relax interrupt test case and give additional time before checking if interrupt handler has been executed.

mprse committed:
lp_us_tickers test - fix count_ticks() function to give the same results on all compilers

count_ticks() function counts ticker ticks elapsed during execution of N cycles of empty while loop.
In current version N value (cycles) is given as volatile paramater in order to disable possible compiler optimalisation.
There was a problem with measured time on different compilers and additionally results on ARM compiler were unexpected (the difference beetween measured elapsed ticks for the same number of cycles was to large). This might be caused by the memory access in order to store updated variable in memory. To fix this issue given numer of cycles has been stored into register and register is decremented (no memory access).

With this fix count_ticks(NUM_OF_CYCLES, 1) call returns 2500 +/-1 for us ticker ticks using each compiler (K64F).

mprse committed:
Enable RTC support for FAMILY_STM32.

mprse committed:
HAL rtc test: check only if rtc_isenabled() returns 1 in case when init is done.

Since some platforms use RTC for low power timer RTC may be already enabled.
Because of that we will only verify if rtc_isenabled() returns 1 in case when init is done and RTC time is set.

mprse committed:
HAL rtc test: increase test timeout.

Verified that on some platforms 30s is not enough.

mprse committed:
Skip higher level ticker tests for targets with stale ticker target specific drivers.

Since target specific ticker drivers are not ready also features which uses ticker in upper layers may not work correctly and tests for these features.
We need to disable also failing higher level ticker related tests (by adding check if DEVICE_USTICKER symbol is available and raise #error [NOT_SUPPORTED] if not).

Note:
tests-mbed_drivers-rtc is new and uses us ticker to perform a delay.

mprse committed:
lp_ticker test - provide lp ticker glitch test case

Test that lp ticker does not glitch backwards due to an incorrectly implemented ripple counter driver.

mprse committed:
lp_us_tickers test: fix `increment test` for slow boards

Increment test proves that ticker counter is incremented by one.
This is done indirectly for high frequency counters where it is impossible to read 'value' and 'value + 1' in two successive ticker reads.
This check is done indirectly by counting ticker ticks elapsed during execution of N cycles of empty while loop. Unfortunately on slow boards with fast tickers like NRF51_DK(16 MHz CPU/1MHz hf ticker) it is possible that for the same N cycles measured number of elapsed ticks in two successive calls is greater than 1. This patch provides fix for such case - measure operation is repeated with the same number of cycles.

mprse committed:
lp_us_tickers test: fix overflow protection

On slow targets with fast high frequency tickers like NRF51_DK (16 Mhz CPU/1MHz ticker) time window for test case execution without overflow needs to be increased. Add parameter to `overflow_protect` function to be able to set different time window without overflow for us ticker and lp ticker.

mprse committed:
Enable RTC support for K64F board.

mprse committed:
K64F: adapt RTC drivers to the new standards (free function)

rtc_free() does not stop the RTC from counting. Only disables clock gate since processor no longer needs to read RTC registers.

mprse committed:
K64F: adapt RTC drivers to the new standards (is_enabled function)

Make rtc_isenabled() to return 1 if the RTC is initialized and the time has been set; 0 otherwise.
Disable clock gate on exit from this function if RTC was initialized.

mprse committed:
tests-mbed_drivers-timerevent: perform test only if USTICKER support is enabled.

mprse committed:
tests-mbed_hal-lp_ticker:  run sleep test only if SLEEP support is enabled.

mprse committed:
tests-mbed_hal-rtc_reset: Add ack from the device after each command is executed

RTC reset test was failing when board has been just powered and RTC reset test was executed for the first time (issue has been detected on CI). In such case RTC initialization takes more time than in futher rtc_init calls. This has impact on green-tea communication. Commands send by host immediately after init command (write, read) were not handled on the device side and no response to the host was provided.
To fix this problem test communication flow has been modified. Device sends ack to the host after RTC init is done. Host sends further RTC commands when ack from the device has been received.

Edit:
There are still some communication issues on the CI. Add ack from the device after each executed command. Increase test timeout and wait after reset.

mprse committed:
Disable RTC support for UBLOX_EVK_ODIN_W2.

This board inherits from STM_FAMILY, but drivers for this board are not consistent with the new standards.
This board needs to be disabled for now.

mprse committed:
lp_us_tickers test: provide minor fixes after review

- count_ticks: fix counter overflow handling,
- count_ticks: use reg_cycles variable in while loop,
- increment test: reduce number of cycles for slow cores if measure process needs to be repeated (difference is greater than 1).

mprse committed:
tests-mbed_drivers-lp_ticker: Adapt tolerance for new Ticker driver

Since this change concerns new ticker driver for NRF51_DK it goas on ticker feature branch.

On master NRF51_DK Ticker driver uses 32kHz RTC for lp and us ticker. Test uses us ticker to measure lp ticker interrupt scheduling (LowPowerTicker.attach), but since the same hardware is used for both tickers the measurement error is constant. On this branch us ticker uses 1 MHz higher precision clock and the results are different - measurement error grows linearly and shows inaccuracy of the RTC.
Test implements constant delta for measured time equal to 2000 us.

Change delta so it depends on lp ticker tested timeout value:
500us for measurement inaccuracy + 5% tolerance for LowPowerTicker

mprse committed:
tests-mbed_drivers-timer: Adapt tolerance for new Ticker driver

Since this change concerns new ticker driver for NRF51_DK it goas on ticker feature branch.

On this branch us ticker is based on 1 MHz higher precision counter (on master is based on 32kHz RTC). As a result of using higher precision counter measurement error increased:
:485::FAIL: Expected 1.060000 Was 1.070478

For delay 1060 ms we measured ~1070 ms so we have about ~1% measurement error which is not bed for this board.

To make it work with 1MHz us ticker increase test tolerance. Define tolerance as follows:
tolerance  = 500us + 0.02 * delay.

bulislaw committed:
Rename DEVICE_LOWPOWERTIMER to DEVICE_LPTICKER

That's to match DEVICE_USTICKER.

bulislaw committed:
Disable Maxim boards as LP_TICKER is used in RTC and they don't respect device_has

bulislaw committed:
Make sure LPTICKER symbols are not used for builds without it for Nordic

bulislaw committed:
Don't use tickless if LPTICKER is not present

The current implementation of tickless requires LPTICKER to be present.

bulislaw committed:
Disable platfroms that fail without LPTICKER

bulislaw committed:
Remove nrf51_dk from supported platform

bulislaw committed:
Rename lp_us tests to common and hf to microsecond

LMESTM committed:
Update STM32 16 bits us_tickers in line with new mbed HAL

The new HAL allows to share the timer bit width and frequency,
the actual handling of mapping 16 bits counter up to 32 bits or
64 bits is now managed by mbed common layer.

This makes this ticker layer very similar to 32bits one and much
easier than before.

mmahadevan108 committed:
MCUXpresso: Enable RTC support

Same MCUXpresso SDK driver used for Kinetis devices

Signed-off-by: Mahesh Mahadevan <mahesh.mahadevan@nxp.com>

LMESTM committed:
Make us_ticker common between 16 and 32 bits counters

mmahadevan108 committed:
MCUXpresso: Enable RTC on LPC54114 and LPC546XX

Signed-off-by: Mahesh Mahadevan <mahesh.mahadevan@nxp.com>

stevew817 committed:
Re-implement RTC for Silicon Labs targets

mbed RTC specifications now dictate that the RTC needs to retain and keep on counting through reset. On Silicon Labs parts, this means the RTC API can not be backed by the Silicon Labs RTC peripheral, since that doesn't provide retention functionality.
Therefore:
* On EFM32GG, EFM32WG, EFM32LG: mbed RTC API is now backed by BURTC.
* On EFM32PG, EFR32MG, EFM32PG12, EFR32MG12: mbed RTC API is now backed by RTCC.
* On EFM32ZG, EFM32HG: mbed RTC API is sadly no longer supported, since these chips don't have retained memory.

stevew817 committed:
Re-implement RTC for Silicon Labs targets

mbed RTC specifications now dictate that the RTC needs to retain and keep on counting through reset. On Silicon Labs parts, this means the RTC API can not be backed by the Silicon Labs RTC peripheral, since that doesn't provide retention functionality.
Therefore:
* On EFM32GG, EFM32WG, EFM32LG: mbed RTC API is now backed by BURTC.
* On EFM32PG, EFR32MG, EFM32PG12, EFR32MG12: mbed RTC API is now backed by RTCC.
* On EFM32ZG, EFM32HG: mbed RTC API is sadly no longer supported, since these chips don't have retained memory.

# Conflicts:
#	targets/TARGET_Silicon_Labs/TARGET_EFM32/lp_ticker.c
#	targets/TARGET_Silicon_Labs/TARGET_EFM32/rtc_api.c
#	targets/targets.json

mprse committed:
lp_us_tickers test: call overflow_protect function in test functions instead test setup handler.

This is done to bypass green-tea setup handler which takes additional time.

bulislaw committed:
Fix rebase error

stevew817 committed:
Re-implement us_ticker and lp_ticker for Silicon Labs targets

Re-implemented both us_ticker and lp_ticker to match the new API and specifications.
Details:
* On EFM32GG, EFM32WG, EFM32LG, EFM32HG, EFM32ZG: Use the RTC peripheral to back lp_ticker, and a TIMER to back us_ticker.
* On EFM32PG, EFR32MG, EFM32PG12, EFR32MG12: Use the RTCC peripheral to back lp_ticker (dual-purpose, also used to back RTC), and a TIMER to back us_ticker.

jeromecoutant committed:
STM32 LPTICKER update for targets supporting LPTIMER

jeromecoutant committed:
LPTICKER targets json update for STM32 supporting LPTIMER

STM32L0, STM32L4, STM32F7 and few STM32F4

mmahadevan108 committed:
MCUXpresso: Update Low Power Ticker

1. Enable LPTICKER for K22, K24, K64, K66, K82, KL82F, KW24D
2. Change the implementation to only use the LPTMR which reduces
   the amount of interrupts generated which is required for tickless
   operation

Signed-off-by: Mahesh Mahadevan <mahesh.mahadevan@nxp.com>

stevew817 committed:
Code style fixes requested by @0xc0170

Tests still passing

mprse committed:
tests-mbed_hal-sleep: add lp/us ticker overflow handling.

mprse committed:
tests-mbed_hal-sleep: use lp ticker data while testing deepsleep.

mprse committed:
tests-mbed_hal-sleep: remove unused variable.

mprse committed:
tests-mbed_hal-sleep: fix formatting.

mprse committed:
tests-mbed_hal-sleep: decrease delay when testing if us ticker is disabled in deep-sleep mode.

This is done because of possible limitations of lowpower ticker freq/width.

mprse committed:
Rename DEVICE_LOWPOWERTIMER to DEVICE_LPTICKER.

mprse committed:
ARM_CM3DS_MPS2: add protection to lp_ticker.c.

Compile this file only when DEVICE_LOWPOWERTIMER is defined.

jeromecoutant committed:
STM32 LPTICKER : enable all STM32 targets supporting the feature

(All STM32 except F1 and F2 families)

jeromecoutant committed:
STM32 LPTICKER : clean include file

TomoYamanaka committed:
Implement of RTC feature for Renesas mbed boards

I implemented the RTC feature.
The mainly changing is here.
- rtc_init()
  Previously, I have initialized the time information register in the function, so the time count was cleaned by every calling rtc_init().
  Currently, rtc_init() doesn't stop RTC from counting, and rtc_init() is safe to call repeatedly.
  Therefore in order to satisfy specifications,I removed the initialization process of the time information register in the function.

- rtc_free()
  Previously, I have initialized the RTC related register same as rtc_init(), so the time count was cleaned by calling rtc_free().
  Currently, rtc_free() doesn't stop RTC from counting.
  Therefore in order to satisfy specifications,I removed the process and decided not to do anything in the function.
  If powerdown the RTC, Supply of the clock to the RTC is stopped, cannot keeping the count.

TomoYamanaka committed:
Implementation of SLEEP feature for Renesas mbed boards

I implemented the SLEEP feature for Rnesas mbed boards.
The mainly changing is here.
- hal_sleep()
  To satisfy the mbed specificationm, I implemented this function newly by using "sleep" that is one of low power mode that is incorporated in our hardware(RZ_A1).
  In the "sleep", peripheral and memory state are maintained, and the peripherals continue to work and can generate interrupts.

- hal_deepsleep()
  To satisfy the mbed specificationm, I implemented this function newly by combined using "sleep" and "module standby" that is the low power mode that is incorporated in our hardware(RZ_A1).
  The "module standby" is peripheral module's powerdown.
  Also in case of our "module standby", it need to read register as dummy when access to each register.

TomoYamanaka committed:
Implementation of USTICKER feature for Renesas mbed boards

I implemented USTICKER feature.

The mainly changing is here.
- I added a macro to mbed_drv_cfg.h for commonalizing code for GR-PEACH and GR-LYCHEE with different clock frequencies, and referenced it's macro at us_ticker.c.
- ticker_init()
  Currently, ticker_init() keep counting, disables the ticker interrupt, and is safe to call repeatedly.
  Therefore, in order to satisfy specifications, I removed GIC_EnableIRQ at end of function and added GIC_DisableIRQ at begin of function.
  When an interrupt is required, it will be set with ticker_set_interrupt().
  If executing the following, the counter has been initialized. So it will not call after executing the first time.
    OSTM1TT   = 0x01;    /* Stop the counter and clears the OSTM1TE bit.     */
    OSTM1TS   = 0x1;    /* Start the counter and sets the OSTM0TE bit.     */

- ticker_free()
  this function stops the counting and powerdown the us_ticker.
  To satisfy the mbed specificationm, I implemented free() function.

- ticker_read()
  Currently, Mbed spec's frequeny is between 250KHz and 8MHz, but the frequency that is used at my ticker is 33MHz.
  Therefore, in order to satisfy specifications, I changed the process to return the timer counter value divided by 32(33MHz / 32).
  Since the calcurate function by using 64 bit is no longer necessay, I removed it.

- ticker_set_interrupt()
  Same as the above read(),
  In order to satisfy specifications, I changed the process to set the value multiplied by 32.

- ticker_fire_interrupt()
  In order to satisfy specifications, I implemented fire_interrupt() function.
  Also I added GIC_EnableIRQ for allowing the interrupt at end of function.

- ticker_get_info()
  To satisfy the mbed specificationm, I implemented ticker_get_info() function. The value of freq includes rounding off.

mprse committed:
Disable SLEEP support for MAX32630FTHR and KW41Z platforms.

These platforms are not consistent with the new sleep standards.

mprse committed:
tests-mbedmicro-rtos-mbed-systimer: execute sleep/deepsleep test cases only when SLEEP support is enabled.

mprse committed:
Disable `tests-mbed_drivers-timerevent` test if USTICKER support is disabled.

mprse committed:
Disable `tests-mbed_drivers-timeout` test if USTICKER support is disabled (NRF52_DK).

c1728p9 committed:
Restore ticker state after each test

Update the ticker common test to clean up after each case by restoring
the ticker IRQ handler. This allows tickers to function normally after
each case has completed.

jeromecoutant committed:
tests-mbed_hal-common_tickers_freq : correct overflowCounter value

jeromecoutant committed:
tests-mbed_hal-common_tickers_freq : use unsigned long

jeromecoutant committed:
tests-mbed_hal-lp_ticker : increase time before deep sleep

mprse committed:
tests-mbed_hal-lp_ticker: Add overflow protection.

Commit:
Rely on ARM's and U-Blox's echo servers

Make the server as default. Use the hostnames instead of plain IP addresses.

jeromecoutant committed:
STM32 LPTICKER : read counter

bulislaw committed:
Disable platforms not supporting new RTC

bulislaw committed:
Rename LOWPOWERTIMER to LPTICKER after merge

Commit:
Starts adding more TCP and UDP performance tests

Starts splitting Greentea test suites to TCP and UDP suites and adds
more test cases like

	tcpsocket_echotest_burst
	tcpsocket_echotest_burst_nonblock
	tcpsocket_endpoint_close
	tcpsocket_recv_100k
	tcpsocket_recv_100k
	tcpsocket_recv_timeout
	tcpsocket_send_timeout
	tcpsocket_thread_per_socket_safety
	udpsocket_echotest_burst
	udpsocket_echotest_burst_nonblock
	udpsocket_recv_timeout
	udpsocket_sendto_timeout

mprse committed:
tests-mbed_hal-common_tickers: increase overflow protection limit

For NR51_DK US_TICKER_OV_LIMIT needs to be increased since if test is run few times in row sometimes fails. This is because NR51_DK is a slow board (16 MHz) with fast and short us ticker counter 1 MHz/16 bits.

Commit:
Add Hardware CRC HAL API specification headers

Define the HAL API header for the Hardware CRC module. This set of functions
allows hardware acceleration of a subset of CRC algorithms for supported
platforms by providing access to the hardware CRC module of certain platforms.

The API is defined as four separate functions:

  - hal_crc_is_supported(polynomial)

    Indicates to the caller if the specific CRC polynomial is supported.

  - hal_crc_compute_partial_start(const uint32_t polynomial)

    Initializes the hardware CRC module with the given polynomial.

  - hal_crc_compute_partial(*data, size)

    Writes an array of bytes to the CRC module to be appended to the calculation

  - hal_crc_get_result()

    Applies the final transformations to the data and returns the result to the
    caller.

jeromecoutant committed:
#6536 rebase correction

jeromecoutant committed:
STM32 LPTICKER update for targets supporting RTC

jeromecoutant committed:
STM32 RTC Init minor update

Commit:
Add K64F Hardware CRC reference implementation


- Add support for all currently defined polynomials
- Add 'CRC' flag to targets.json for K64F profile

Commit:
Adapt MbedCRC class to support Hardware CRC 

- Move CRC polynomial enum into HAL layer, so it's accessible from platform
  implementations

- Add enum to CRC class to indicate which mode the CRC class should use:
  HARDWARE, TABLE, or BITWISE

- Add calls to HAL Hardware CRC API to each of the compute functions when the
  class is in HARDWARE mode.

- Add missing constructor call to template constructor, and remove const from
  delegating constructor.

Commit:
Fix CRC class coding conventions

Commit:
Remove support for 7/8 bit CRC polynomials for K64F

Commit:
HAL CRC: Return early when calling compute with null or 0 size buffer

Commit:
Add CRC configuration options to HAL API

Commit:
mbed_ticker_api remove specific case for 32768 in favor of all frequencies divisible by 2

Commit:
Changed struct position to try to maintain binary compatibility
Changed comment because a lot of numbers are divisible by 2

Commit:
Updated types according to header changes

juhaylinen committed:
Update Greentea WiFi tests

Use similar network interface configuration as used in netsocket tests
Improve test coverage

Commit:
HAL CRC: Add bit width parameter to crc_config_t

Commit:
Adds more TCP and UDP test cases

Adds TCP and UDP test cases

        tcpsocket_open_limit
        udpsocket_open_limit

Commit:
Add support for final_xor for K64F Hardware CRC

Commit:
HAL CRC: Fix code to match coding conventions

Commit:
Adds TCP and UDP test cases

	tcpsocket_open_close_repeat
	udpsocket_open_close_repeat

juhaylinen committed:
Update Greentea test configuration files

Add .json files for WicedInterface and SpwfSAInterface
Add target configurations for
MTB_UBLOX_ODIN_W2
MTB_ADV_WISE_1530
MTB_USI_WM_BN_BM_22
MTB_MXCHIP_EMW3166
NUCLEO_F401RE

Commit:
Adds TCP test case

                tcpsocket_connect_invalid

Commit:
Adds TCP test case

	tcpsocket_echotest
	tcpsocket_echotest_nonblock

Commit:
Removes obsolete TCP and UDP Greentea test cases

Commit:
Adds UDP Greentea test case

	udpsocket_sendto_invalid

Commit:
HAL CRC: K64F: Set output width based on input width, not the polynomial

Commit:
Adds TCP test case

	tcpsocket_send_repeat

Commit:
Adds UDP Greentea test case

        udpsocket_sendto_repeat

Commit:
HAL CRC: Add DEVICE_CRC macro to doxygen files

Commit:
HAL CRC: Update function documentation with new config parameter

Commit:
HAL CRC: Fix inverted CRC mode selection

Commit:
HAL CRC: Add additional thread safety notes to the API

Commit:
HAL CRC: Basic implementation for STM32F0

Commit:
Adds UDP Greentea test cases

	udpsocket_echotest
	udpsocket_echotest_nonblock

Commit:
HAL CRC: Use HAL polynomial enum instead of STM32 enum

Commit:
HAL CRC: Move STM32 implementation to common folder and add targets

Commit:
Greentea TCP test cases - check all socket calls

Commit:
HAL CRC: Add additional STM32 targets

bulislaw committed:
Disable failig Nordic targets

mmahadevan108 committed:
MCUXpresso: Update the usticker implementation

Enabled usticker for K22F, K24F, K66F, K82F, KL82Z, KW24D

Signed-off-by: Mahesh Mahadevan <mahesh.mahadevan@nxp.com>

mmahadevan108 committed:
MCUXpresso: Enable usticker for LPC546XX and LPC54114

Signed-off-by: Mahesh Mahadevan <mahesh.mahadevan@nxp.com>

mmahadevan108 committed:
MCUXpresso: Enable usticker on MIMXRT1050_EVK

Signed-off-by: Mahesh Mahadevan <mahesh.mahadevan@nxp.com>

Commit:
Fixes Greentea UDP test cases

	udpsocket_echotest.cpp
	udpsocket_echotest_burst.cpp

bulislaw committed:
Remove sleep from rf52 due to failig tests

bulislaw committed:
Disable MAX targets failig to build

Commit:
Cellular: Fix AT URC handler not to dismiss incoming chars

Commit:
Switches Greentea netsocket test case naming convention

From lowercase to uppercase and drops 'test_'-prefix. Test case
descriptions do also match to TC names now.

Additionally all the TCs are not behing MBED_EXTENDED_TESTS
anymore.

mprse committed:
Move common_rtc.c and us_ticker.h files to the valid directory.

I decided to move these files to the targets/TARGET_NORDIC/TARGET_NRF5x/TARGET_NRF51 since us_ticker.h is for sure specific for NRF51_DK and common_rtc.c might be valid also for NRF52, but this needs to be checked while porting NRF52_DK board.

mprse committed:
Enable support for NRF51 boards.

mprse committed:
tests-events-queue: modify tolerance to percentage form

On some boards without SYSTICK support lp ticker is used to trace system ticks.
Event queue uses system ticker to count delay when callback is to be fired.
New ticker implementation for NRF51_DK provides fast and accurate counter for us ticker which shows inaccuracy of the system ticker(which is based on slow and inaccurate lp ticker). This is why measured error grows linearly and percentage form needs to be used.

Currently we use constant tolerance value equal to: 5 [ms].
This patch modifies tolerance to the following form: 5% of measured time [ms].

In this test the lowest delay is equal to 100 [ms], so 5% corresponds to 5 [ms] tolerance - original value. This means that min tolerance is 5 [ms] so this change should not have negative impact on other targets.

juhaylinen committed:
Fix Greentea WiFi test build issues

Commit:
Fixes findings from Greentea netsocket UDP test cases

Commit:
Fixes findings from Greentea netsocket TCP test cases

mmahadevan108 committed:
KW41Z: Update usticker to use TPM instead of LPTMR for ticker

This will free up the LPTMR to be used as lpticker

Signed-off-by: Mahesh Mahadevan <mahesh.mahadevan@nxp.com>

mmahadevan108 committed:
MIMXRT1050_EVK: Update lpticker implementation

Use only the GPT module and avoid using RTC.

Signed-off-by: Mahesh Mahadevan <mahesh.mahadevan@nxp.com>

mprse committed:
tests-mbed_drivers-timerevent: Disable deep-sleep mode in TestTimerEvent class.

Since now us ticker is disabled in deep-sleep mode and tests-mbed_drivers-timerevent verifies TimerEvent class which uses us ticker then deep-sleep mode needs to be disabled during the test execution.

mprse committed:
tests-mbed_drivers-lp_ticker: increase tolerance for NRF51_DK

This test fails sometimes while testing 1 ms timeout on NRF51_DK board.
This board is very slow and lp ticker is inaccurate.
I noticed that measured results are very close to the tolerance value. For 1ms delay we measure in failure case 1560 us and tolerance is set to 550.
Since measurement method is also significantly flawed (extra time for setting up ticker and execution of a ticker irq handler) I recommend to increase tolerance by least +100 us for now.

mprse committed:
Disable us ticker in deep-sleep mode.

mmahadevan108 committed:
KL05, KL25, KL26, KL46: Enable usticker

Signed-off-by: Mahesh Mahadevan <mahesh.mahadevan@nxp.com>

mmahadevan108 committed:
LPC1768: Enable usticker

Signed-off-by: Mahesh Mahadevan <mahesh.mahadevan@nxp.com>

jessexm committed:
mbed-os v5.9 RTC implementation

jessexm committed:
mbed-os v5.9 USTICKER implementation

jessexm committed:
mbed-os v5.9 LPTICKER implementation

Commit:
ADI: Re-enable DEVICE_RTC for EV_COG_AD3029LZ and EV_COG_AD4050LZ

mprse committed:
Enable support for NRF51 boards.

Make it consistent with master.

mprse committed:
tests-mbed_hal-common_tickers: disable os tick while running test cases

It has been noticed on NRF51_DK board that occurrence of system tick have impact on test execution.
NRF51_DK is a slow board with fast us ticker and handling of system tick interrupt takes about 250 us ticker ticks which have huge influence on interrupt tests where interrupt is set to <current count> + 100, 200, 500 ticks.

Commit:
Remove trailing spaces of adi_rtc.c

Commit:
Preserve RTC counter value while re-initialization

mmahadevan108 committed:
KL43Z: Update usticker to use TPM instead of LPTMR for ticker

This will free up the LPTMR to be used as lpticker

Signed-off-by: Mahesh Mahadevan <mahesh.mahadevan@nxp.com>

mmahadevan108 committed:
KL27Z: Update usticker to use TPM instead of LPTMR for ticker

This will free up the LPTMR to be used as lpticker

Signed-off-by: Mahesh Mahadevan <mahesh.mahadevan@nxp.com>

mmahadevan108 committed:
KW41Z, KL43Z, KL27Z: Enable LPTICKER

Signed-off-by: Mahesh Mahadevan <mahesh.mahadevan@nxp.com>

Commit:
Fix us_ticker.c to support "USTICKER"

- Disable microsecond ticker interrupt on reinitialization
- Skip us_ticker_set_interrupt() if timestamp is already past
- Eliminate tmr2Config since tmrConfig is adequate for all timer config

mprse committed:
NRF52_DK: use new ticker driver designed for NRF51_DK

mprse committed:
NRF52_DK: Enable lp ticker and us ticker support

mprse committed:
tests-mbed_hal-common_tickers: fix bug in ticker increment test case

It is possible that the difference between base and next tick count on some platforms is greater than 1, in this case we need to repeat counting with the reduced number of cycles (for slower boards).
If number of cycles has been reduced than base tick count needs to be redefined. This operation is missing and is added by this patch.

mprse committed:
tests-mbed_drivers-timerevent: increase tolerance for NRF52_DK

Test inserts event into the TimerEvent object which is scheduled 50 ms in the future. Then test thread hangs on the semaphore for 51 ms and after that time we expect that event handler has been executed. This test fails sometimes on NRF51_DK, NRF52_DK since different clocks are used for event handling and delay. TimerEvent class uses fast us ticker and semaphores are based on lp ticker (NRF51_DK) or System Tick Timer (NRF52_DK).
We assume that ticker measurement error is 10% and our 1 [ms] extra corresponds to 2% error. I suggest to increase delay to 2 [ms] (4%). This should be enough for all boards at this moment.

mprse committed:
tests-mbed_drivers-lp_timer: modify tolerance to percentage form

Represent tolerance value as it is done in `tests-mbed_drivers_timer`.
Use 5% tolerance for lp ticker.

mprse committed:
tests-mbed_hal-flash: Remove 5% extra tolerance for NRF52_DK

NRF52_DK is now based on fast and accurate 1MHz counter.

mprse committed:
tests-mbed_hal-flash: optimieze time_cpu_cycles() function

This function should perform instruction cycles count in critical section.
Additionally remove redundant code.

mprse committed:
NRF52_DK: disable us ticker in deep-sleep mode

This functionality is required by new sleep standards.

Commit:
Adding checks for stack initialization

Metadata APIs should return an error if the stack is not initialized yet.

M-ichae-l committed:
fix realtek_rtl8195am IPv6 build fails

remove the "dhcps.h" and "dhcps.c" files
edit "wifi_conf.c" and "rtw_emac.cpp" to fix compile error
coding style fix

kjbracey reviewed (changes requested):
I don't think we can take any ethernet-related changes into master at the moment while feature-emac is pending.

M-ichae-l commented:
@kjbracey-arm 
OK. We will check again after the feature-emac merged with master. Thank you.

Commit:
Corrected asynchronous DNS functionality

- Set network stack to store event queue so that mbed::mbed_event_queue() call
  is not needed every time call_in() is called
- Added dns state variables and enum (states are: created, initiated and cancelled)
- Corrected DNS response handling so that if DNS server returns that host name is
  unknown the DNS query is not tried again
- Reorder mutexes in nsapi_dns_query_multiple_async()
- Created nsapi_dns_query_async_initiate_next() function to initiate the next
  DNS query from the queue after delete of previous query
- Added dsn_timer_running variable to supervise DNS timer start/stop
- Changed cancel function to only mark query as deleted and moved deletion
  to timer function. This allows to run socket close on DNS thread
- Added new nsapi error NSAPI_ERROR_TIMEOUT for DNS (and other) timeouts

mprse committed:
Make common_rtc.c compatible with SDK 14.2 and move it common dir

mprse committed:
Enable us lp ticker support for entire MCU_NRF52832 family

Commit:
Copy edit crc_api.h

Copy edit file, mostly for American spelling and consistent tense across files.

bulislaw committed:
Change LPTICKER spec to allow clock freq starting at 4kHz

Commit:
Adding usticker to MTB_ODIN

mprse committed:
NRF52840_DK: Enable lp us ticker support

Commit:
Stability improvements & CRYPTO_ERROR addition

General stability improvements are performed.
A flag is added if a Class C RX2 window is open.
We shouldn't open it again if its already opened.

TX_CRYPTO_ERROR is renamed to CRYPTO_ERROR.
Keeping TX_CRYPTO_ERROR for backwards compatibility.

kjbracey committed:
Merge pull request #6847 from ARMmbed/feature-emac

Merge feature-emac branch into master

Commit:
Merge pull request #6794 from RonEld/cc310_porting

Cryptocell 310 support

Commit:
Merge pull request #6642 from yossi2le/yossi-device-key-driver

Device key implementation

Commit:
Merge pull request #6932 from paul-szczepanek-arm/security-manager-dev

BLE privacy, signing, persistent security database

bulislaw committed:
Rename new occurrences of LOWPOWERTIMER to LPTICKER

Commit:
Merge pull request #6983 from SenRamakri/sen_ErrorHandling_Push2

Standardized Error Handling and Error Codes

Commit:
Merge pull request #6927 from AriParkkila/cellular-debug

Cellular: AT debugging improved

Commit:
Merge pull request #6433 from cmonr/python3-in-travisci

Enabl-able Python 3 tools testing in Travis CI

mharringADI committed:
Fixed incorrect clean_flags method and float-abi linker option for CCES exporter.

bulislaw committed:
Fix missmatch between feature branches and new CMSIS

Commit:
Limit devicekey example to only K64F

Lines up devicekey example with nvstore example

Commit:
Enable DeviceKey class reference for Doxygen

Add conditional, so Doxygen renders properly.

Commit:
Picking a deprecation warning and moving forward

bulislaw committed:
Fix target definition for Odin boards

KariHaapalehto committed:
Add check for _interface

Commit:
Fix build for MBT_ODIN_W2 and MBED_CONNECT_ODIN when using WiFi

Odin WiFi driver was only supplied for EVK board.
These all are just the same module, so using the main target name
that all inherit.

0xc0170 committed:
maxim: fix rtc if lp ticker not defined

lp ticker handler defined only if lp ticker is defined. In case not,
use only overflow, rtc0 not used.

0xc0170 committed:
efm32: remove unused undefined function from rtc

Commit:
Made lwip memory configuration more specific for NXP targets

There is a configuration problem with UBLOX_C027 cellular because of missing
LWIP memory configuration for the target.

Added configuration for LPC1768, LPC4088, LPC4088_DM targets and changed
NXP to more specific LPC546XX.

0xc0170 committed:
max32600: fix lp ticker functions - do not inline

Wrong linkage specified, causing build to fail.

0xc0170 committed:
Merge pull request #7003 from adbridge/master

Add devicekey example to examples list

Commit:
Added STM32L433RC to iar exporter options

mprse committed:
Fix tolerance definition in ticker tests.

Commit:
Merge pull request #7000 from kfnta/uvisor_depr

Add warning about FEATURE_UVISOR being deprecated

Commit:
Merge pull request #7012 from ARMmbed/AnotherButler-patch-1

Enable DeviceKey class reference for Doxygen

0xc0170 committed:
Merge pull request #7015 from SeppoTakalo/fix_odin

Fix build for MBT_ODIN_W2 and MBED_CONNECT_ODIN when using WiFi

Commit:
Fix array overflow in gpio configuration code for NRF5x

Reintroduce PR #6021

https://github.com/ARMmbed/mbed-os/pull/6021

which was accidentally removed by PR #6711

https://github.com/ARMmbed/mbed-os/pull/6711

Commit:
Merge pull request #6708 from scartmell-arm/feature-hal-hardware-crc

Add Hardware CRC HAL API specification

bulislaw committed:
Rename LOWPOWERTIMER to LPTICKER

bulislaw committed:
NRF52 is not compatible with new RTC HAL spec

Commit:
Merge pull request #6910 from hasnainvirk/metadata

LoRaWAN: Adding acquisition of metadata, backoff and a cancel_send() API

bulislaw committed:
Sleep was not ported to MAX32630FTHR

adbridge committed:
Add new examples to example list for CI.

The following examples have been added:

https://github.com/ARMmbed/mbed-os-example-thread-statistics
https://github.com/ARMmbed/mbed-os-example-sys-info
https://github.com/ARMmbed/mbed-os-example-cpu-usage
https://github.com/ARMmbed/mbed-os-example-cpu-stats

adbridge committed:
Add 2 more examples and remove deprecated one.

Removed:
mbed-os-example-fat-filesystem
Added:
mbed-os-example-error-handling
mbed-os-example-blockdevice

Commit:
Merge pull request #7019 from mikaleppanen/lwip_lpc_mem_conf

Made lwip memory configuration more specific for NXP targets

bulislaw committed:
Fix missmatch between targets.son

Commit:
Merge pull request #6665 from VeijoPesonen/greentea_netsocket_more_tests

Greentea netsocket more tests

Commit:
Updated blockdevice example to target only K64F

Commit:
Merge pull request #6981 from yogpan01/master

Remove ns_event_loop_thread_start() from NDInterface and ThreadInterface connect()

Commit:
Merge pull request #7023 from adbridge/master

Add new examples to example list for CI.

bulislaw committed:
Disable RTC on MAX32630FTHR due to failing tests

bulislaw committed:
Disable intermittently failing sleep test

bulislaw committed:
Disable FLASH for AD boards as they fail the tests intermittently

bulislaw committed:
Disable FLASH algo for AD boards

bulislaw committed:
Disable sleep tests as they are unstable

kjbracey committed:
Fix lwIP PPP glue

EMAC changes broke PPP glue - fix it up. In particular, PPP steals
the netif->state pointer, so we can't use it as a back pointer to
our LWIP::Interface.

mprse committed:
tests-mbed_hal-common_tickers: Fix increment test case implementation

In case when base tick count is different than next tick count check first if the difference is equal to 1 tick (this is what we are looking for) if not then decrease the tick count.

Repeat counting process few times before incrementing the number of cycles in case when base tick count is equal to next tick count.

TomoYamanaka committed:
Fix redeclaration of type name "bool_t"

I fixed redeclaration of type name "bool_t" for target Renesas because this typedef has been defined in rtx_core_ca.h by #6273.

0xc0170 committed:
Merge pull request #7009 from bulislaw/merge_feature_branches_for_5.9

Bring in improved HAL APIs to master

0xc0170 committed:
Merge pull request #7030 from kjbracey-arm/emac_ppp_fix

Fix lwIP PPP glue

SeppoTakalo commented:
@M-ichae-l Any update on this? EMAC has now been merged in and IPv6 build is still broken.

Please rebase/update your changes on top of master.

Commit:
Merge pull request #6928 from jarvte/fix_cellular_connect_disconnect

Cellular: Fixed con-disc sequence can now be called multiple times.

Commit:
Merge pull request #6947 from evva-sfw/hotfix/EFM32PG12_STK3402_Expansion_Header

EFM32PG12_STK3402 - Correct wrong Expansion-Header pin namings

Commit:
Merge pull request #6965 from mirelachirica/fix_unittests

Cellular: Unit tests fixes

Commit:
Merge pull request #6970 from stevew817/efm32_flash_bugfix

Fixes to EFM32 port

Commit:
Merge pull request #6984 from stevew817/'efm32_irq_bugfix'

EFM32 IRQ handling fix

Commit:
Merge pull request #6991 from amq/efm32-universal-rtx

EFM32: make mbed_rtx.h depend of families instead of targets

Commit:
Merge pull request #7011 from mharringADI/fix-cces-export-float-abi

Fixed float-abi linker option for CCES exporter.

Commit:
Merge pull request #7014 from KariHaapalehto/interface_check

Add check for _interface

Commit:
Merge pull request #7020 from marcemmers/iar-add-stm32l433

Add STM32L433RC to IAR exporter definitions

Commit:
Merge pull request #7022 from marcuschangarm/fix-gpio

Fix array overflow in gpio configuration code for NRF5x

Commit:
Merge pull request #7033 from AriParkkila/cellular-at-fix

Cellular: Fix AT URC handler not to dismiss incoming chars

Commit:
Merge pull request #7034 from TomoYamanaka/master

Fix redeclaration of type name "bool_t" for Renesas

Commit:
Merge pull request #7035 from mprse/ticker_fix

tests-mbed_hal-common_tickers: Fix increment test case implementation

Commit:
Merge pull request #7016 from mikaleppanen/async_dns_corr

Corrected asynchronous DNS functionality

M-ichae-l committed:
Merge branch 'fix-realtek_rtl8195am-IPv6-build-fails' of https://github.com/M-ichae-l/mbed-os into fix-realtek_rtl8195am-IPv6-build-fails

M-ichae-l committed:
fix realtek_rtl8195am IPv6 build fails update

rebase for master branch
coding style fix

M-ichae-l commented:
@SeppoTakalo 
I have just rebased the branch and updated some files.

M-ichae-l commented:
It is looks like that the rebase has made too much commits. So I have created a new PR for rebase and update. We can close this PR. 
Please refer to #7056

SeppoTakalo commented:
@M-ichae-l Please close this PR as it is now replaced with #7055
